org.adempiere.base/src/org/compiere/acct/Doc.java

```diff
@@ -1303,8 +1303,10 @@ public int getValidCombination_ID (int AcctType, MAcctSchema as)
 			int cmp = getAmount(AMTTYPE_Charge).compareTo(Env.ZERO);
 			if (cmp == 0)
 				return 0;
-			else
+			else if (cmp < 0)
 				sql = "SELECT CH_Expense_Acct FROM C_Charge_Acct WHERE C_Charge_ID=? AND C_AcctSchema_ID=?";
+			else
+				sql = "SELECT CH_Revenue_Acct FROM C_Charge_Acct WHERE C_Charge_ID=? AND C_AcctSchema_ID=?";

```

Comment:
is this by sign logic always the safe thing to do ? perhaps a potential problem with reversal ?

Reply:
ideally, we need unit test to validate the normal and corner cases